packages/vaadin-upload/theme/material/vaadin-upload-styles.js

```diff
@@ -206,6 +194,38 @@ registerStyles(
       content: var(--material-icons-clear);
     }
 
+    [part$='button']::after {
+      position: absolute;
+      content: '';
+      top: 0;
+      left: 0;
+      width: 40px;
+      height: 40px;
+      border-radius: 50%;
+      background-color: var(--material-body-text-color);
+      transform: scale(0);
+      opacity: 0;
+      transition: transform 0.08s, opacity 0.01s;
+      will-change: transform, opacity;
+    }
+
+    [part$='button']:hover::after {
+      opacity: 0.08;
+    }
+
+    [part$='button']:focus::after {
+      opacity: 0.12;
+    }
+
+    [part$='button']:active::after {
```

Comment:
Using the `:active` pseudo-class may not be enough as there is an issue https://github.com/vaadin/vaadin-button/issues/4 and, as for `vaadin-button`, it uses the special `ActiveMixin` that toggles the `active` attribute.